opds_import.py

```diff
@@ -321,26 +464,39 @@ def combine(self, d1, d2):
                 new_dict[k] = v
         return new_dict
 
+
     @classmethod
-    def extract_metadata_from_feedparser(cls, feed):
+    def extract_data_from_feedparser(cls, feed, data_source):
         feedparser_parsed = feedparser.parse(feed)
-        values = {}
+        values_meta = {}
+        values_circ = {}
         status_messages = {}
         for entry in feedparser_parsed['entries']:
-            identifier, detail, status_message = cls.detail_for_feedparser_entry(entry)
+            identifier, detail_meta, detail_circ, status_message = cls.data_detail_for_feedparser_entry(entry=entry, data_source=data_source)
+
             if identifier:
-                if detail:
-                    values[identifier] = detail
                 if status_message:
                     status_messages[identifier] = status_message
+                else:
+                    if detail_meta:
+                        values_meta[identifier] = detail_meta
+                    if detail_circ:
+                        values_circ[identifier] = detail_circ
+            else:
+                # That's bad. Can't make an item-specific error message, but write to 
+                # log that something very wrong happened.
+                logging.error("Tried to parse an element without a valid identifier.")
```

Comment:
I don't think this is bad enough to warrant an `.error`. It looks like this happens if someone serves an OPDS entry that lacks an <id> tag, which is annoying but not fatal to us. I think warn() is sufficient, and it would also be good to give a string representation of the entry.